### PR TITLE
adds haml, adds release packager

### PR DESCRIPTION
**Pull request by nii236**

### CODE REVIEW

**Comment by nii236** [comment]
Need to fix up travis to work with this as well

File: config/prod.exs [hunk 1]
```diff
@@ -13,16 +13,18 @@ use Mix.Config
...
+  root: ".",
+  version: Mix.Project.config[:version]
 
 # Configure your database
 config :noot, Noot.Repo,
   adapter: Ecto.Adapters.Postgres,
   url: System.get_env("DATABASE_URL"),
   pool_size: String.to_integer(System.get_env("POOL_SIZE") || "10"),
-  ssl: true
+  ssl: false
```
Comment (jgunnink) [inline config/prod.exs:27]: It's enabled on staging, why disable here?

File: mix.exs [hunk 2]
```diff
@@ -36,9 +36,11 @@ defmodule Noot.Mixfile do
      {:postgrex, ">= 0.0.0"},
      {:phoenix_html, "~> 2.6"},
      {:phoenix_live_reload, "~> 1.0", only: :dev},
+     {:phoenix_haml, "~> 0.2.1"},
      {:gettext, "~> 0.11"},
      {:cowboy, "~> 1.0"},
-     {:comeonin, "~> 2.3"}]
+     {:comeonin, "~> 2.3"},
+     {:distillery, "~> 0.10"}]
```
Comment (jgunnink) [inline mix.exs:43]: 👍

File: rel/config.exs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+# You may define one or more environments in this file,
+# an environment's settings will override those of a release
+# when building in that environment, this combination of release
+# and environment configuration is called a profile
+
+environment :dev do
+  set dev_mode: true
+  set include_erts: false
+  set cookie: :"JFYyyN/w(k~X4~~eDY`}V{PU|/V_BNF^y=u4O{I*Ez`A~FYxMXYGZ/q_edh22[L+"
```
Comment (jgunnink) [inline rel/config.exs]: My cookie fu is not super strong - should this be an environment var?

File: web/templates/layout/app.html.haml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+          Logged in as
+          %strong= user.username
+        = link "Logout", to: session_path(@conn, :delete, user.id), method: :delete, class: "link dim gray f6 f5-ns dib mr3"
+      - else
+        = link "Login", to: session_path(@conn, :new), class: "link dim gray f6 f5-ns dib mr3"
+    .measure-wide.center.pa4
+      %p{role: "alert"}= get_flash(@conn, :info)
+      %p{role: "alert"}= get_flash(@conn, :error)
+      %main{role: "main"}= render @view_module, @view_template, assigns
+    / /container
```
Comment (jgunnink) [inline web/templates/layout/app.html.haml]: Remove

File: web/templates/session/form.html.haml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        - for {attr, message} <- f.errors do
+          %li
+            = humanize(attr)
+            = message
+  .form-group
+    = text_input f, :username, class: "form-control", placeholder: "Username"
+  .form-group
+    = password_input f, :password, class: "form-control", placeholder: "Password"
+  .form-group
+    = submit "Submit", class: "pa3 f4 br2 bg-white ba b--black center db w-third"
```
Comment (jgunnink) [inline web/templates/session/form.html.haml]: can you do this?

``` haml
.form-group
  = text_input f, :username, class: "form-control", placeholder: "Username"
  = password_input f, :password, class: "form-control", placeholder: "Password"
  .form-actions
    = submit "Submit", class: "pa3 f4 br2 bg-white ba b--black center db w-third"
```

File: web/templates/session/new.html.haml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+%h2.tc Login
+
```
Comment (jgunnink) [inline web/templates/session/new.html.haml]: bonus ws

File: web/templates/user/form.html.haml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    = error_tag f, :email
+  .form-group
+    = label f, :password, "Password", class: "control-label"
+    = password_input f, :password, class: "form-control"
+    = error_tag f, :password
+  .form-group
+    = label f, :password_confirmation, "Password Confirmation", class: "control-label"
+    = password_input f, :password_confirmation, class: "form-control"
+    = error_tag f, :password_confirmation
+  .form-group
```
Comment (jgunnink) [inline web/templates/user/form.html.haml:17]: `.form-actions` ?
or
`.form-actions form-group` ?

File: web/templates/user/index.html.haml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+%h2 Listing users
+  %table.table
```
Comment (jgunnink) [inline web/templates/user/index.html.haml]: duplicated?

File: web/templates/user/show.html.haml [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+%ul
+  %li
+    %strong Username:
+    = @user.username
+  %li
+    %strong Email:
+    = @user.email
+  %li
+    %strong Password digest:
+    = @user.password_digest
```
Comment (jgunnink) [inline web/templates/user/show.html.haml]: We don't need the digest. Just the user anme and email field should be good.

**Review by jgunnink — CHANGES_REQUESTED** [review]
I think you may be working from an old version of develop. For example, I took out the password digest on the table, which is still here.

File: config/prod.exs (hunk 1 above)
Comment (nii236) [reply config/prod.exs:27]: Because the prod server doesn't have SSL certs yet

File: web/templates/user/index.html.haml (hunk 8 above)
Comment (nii236) [reply web/templates/user/index.html.haml]: Nope, bootstrap class

File: web/templates/user/form.html.haml (hunk 7 above)
Comment (nii236) [reply web/templates/user/form.html.haml:17]: There is no form-group class

**Review by jgunnink — APPROVED** [review]
Looks good. Just needs screenshots 👍

**Comment by nii236** [comment]
What for?

**Pull request merged**